### ISSUE TITLE
No components inside app.js creates a flicker on refresh

### ISSUE THREAD

**Issue by matthewharwood** [preactjs/preact-cli#785]
**Do you want to request a _feature_ or report a _bug_?**
Bug

**What is the current behaviour?**
When refreshing `preact` project there shall not be a flicker. 

*Note this only happens on ^3.0.0 cli.*

**If the current behaviour is a bug, please provide the steps to reproduce.**

*Install/scaffold:*
```
npm install -g preact-cli
preact create default playground
cd playground
npm uninstall preact-cli --save-dev && npm install preact-cli@3.0.0-next.19
```
*Edit:*
- Remove the `<Header />` component from the `src/components/app.js`
- `npm start`
 
![flicker](https://imgur.com/QX6YFfJ.gif)

**app.js**

```
	render() {
		return (
			<div id="app">
				<Router onChange={this.handleRoute}>
					<Home path="/" />
					<Profile path="/profile/" user="me" />
					<Profile path="/profile/:user" />
				</Router>
			</div>
		);
	}
```

**What is the expected behaviour?**
No flicker if there is only a Router in the app.js

**Please mention other relevant information.**

- Node version: 11.0.0
- npm version: 6.4.1
- Operating system OSX 10.14.4
- CLI version: `preact-cli@3.0.0-next.19`
- Browser: Chrome

**Comment by psabharwal123** [comment]
i am experiencing this too, seems like https://github.com/preactjs/preact-cli/issues/281 is back? @lukeed

**Comment by psabharwal123** [comment]
i tried to debug, the issue is in @preact/async-loader - both `this.nextBase and this.__b` are always undefined, also in `this.render = (p, s) => (s.child ? h(s.child, p) : old); ` during the initial render s = {} and old is undefined. the code 
`old = h(b.nodeName, {
						dangerouslySetInnerHTML: { __html: b.innerHTML },
					});`
never gets called

**Comment by psabharwal123** [comment]
also tagging @prateekbh if you have any ideas as to how this should be fixed.

**Comment by ForsakenHarmony** [comment]
async loader is broken with preact 10, it's the last thing needed for the full release of cli v3.0

**Comment by psabharwal123** [comment]
Good to know, thanks!

**Comment by rackstar17** [comment]
@ForsakenHarmony thanks for the update, I'm using preact-cli for my next project and it has the same flickr issue. Can we expect the fix for the same in some upcoming release soon ? @lukeed

**Comment by reznord** [comment]
This is fixed in the latest [3.0.0-rc.16](https://github.com/preactjs/preact-cli/tree/3.0.0-rc.16)
